projects/SDE Lighthouse 03 - Kontopantelis/template-sql/5.medications.template.sql

```diff
@@ -0,0 +1,181 @@
+        WHEN concept='trazodone' THEN 'ACB1'
+        WHEN concept='triamterene' THEN 'ACB1'
+        WHEN concept='warfarin' THEN 'ACB1'
+        WHEN concept='amantadine' THEN 'ACB2'
+        WHEN concept='belladona' THEN 'ACB2'
+        WHEN concept='carbamazepine' THEN 'ACB2'
+        WHEN concept='cyclobenzaprine' THEN 'ACB2'
+        WHEN concept='cyproheptadine' THEN 'ACB2'
+        WHEN concept='loxapine' THEN 'ACB2'
+        WHEN concept='pethidine' THEN 'ACB2'
+        WHEN concept='levomepromazine' THEN 'ACB2'
+        WHEN concept='oxcarbazepine' THEN 'ACB2'
+        WHEN concept='pimozide' THEN 'ACB2'
+        WHEN concept='amitriptyline' THEN 'ACB3'
+        WHEN concept='amoxapine' THEN 'ACB3'
+        WHEN concept='atropine' THEN 'ACB3'
+        WHEN concept='benztropine' THEN 'ACB3'
+        WHEN concept='chlorphenamine' THEN 'ACB3'
+        WHEN concept='chlorpromazine' THEN 'ACB3'
+        WHEN concept='clemastine' THEN 'ACB3'
+        WHEN concept='clomipramine' THEN 'ACB3'
+        WHEN concept='clozapine' THEN 'ACB3'
+        WHEN concept='darifenacin' THEN 'ACB3'
+        WHEN concept='desipramine' THEN 'ACB3'
+        WHEN concept='dicyclomine' THEN 'ACB3'
+        WHEN concept='diphenhydramine' THEN 'ACB3'
+        WHEN concept='doxepin' THEN 'ACB3'
+        WHEN concept='flavoxate' THEN 'ACB3'
+        WHEN concept='hydroxyzine' THEN 'ACB3'
+        WHEN concept='imipramine' THEN 'ACB3'
+        WHEN concept='meclozine' THEN 'ACB3'
+        WHEN concept='mepyramine' THEN 'ACB3'
+        WHEN concept='nortriptyline' THEN 'ACB3'
+        WHEN concept='orphenadrine' THEN 'ACB3'
+        WHEN concept='oxybutynin' THEN 'ACB3'
+        WHEN concept='paroxetine' THEN 'ACB3'
+        WHEN concept='perphenazine' THEN 'ACB3'
+        WHEN concept='procyclidine' THEN 'ACB3'
+        WHEN concept='promazine' THEN 'ACB3'
+        WHEN concept='promethazine' THEN 'ACB3'
+        WHEN concept='propantheline' THEN 'ACB3'
+        WHEN concept='scopolamine' THEN 'ACB3'
+        WHEN concept='solifenacin' THEN 'ACB3'
+        WHEN concept='tolterodine' THEN 'ACB3'
+        WHEN concept='trifluoperazine' THEN 'ACB3'
+        WHEN concept='trihexyphenidyl' THEN 'ACB3'
+        WHEN concept='trimipramine' THEN 'ACB3'
+        WHEN concept='trospium' THEN 'ACB3'
+    END AS "MedicationGroup",
+    concept AS "MedicationType",
+    description AS "Medication"
+FROM LH003_Medication_Codes x
+LEFT OUTER JOIN {{code-set-table}} c 
+ON c.code = x."SuppliedCode"
+WHERE MedicationDate >= '2006-01-01'
+UNION
+-- Link the above to the data from the Refset clusters
+SELECT GmPseudo, TO_DATE("MedicationDate") AS "MedicationDate", 
+    CASE
+        WHEN "Field_ID" = 'BENZODRUG_COD' THEN 'Benzodiazipine related'
+        WHEN "Field_ID" = 'ANTIPSYDRUG_COD' THEN 'Antipsychotic'
+    END AS "MedicationCategory",
+		-- to get the medication the quickest way is to just split the description by ' ' and take the first part
```

Comment:
It almost always works. Codes like "Product containing..." aren't typically used - they are just nodes in the SNOMED hierarchy but not actually things that can be prescribed. The logic shouldn't be used without checking though because you do occasionally get medications that don't start with the drug name. I've mainly found it with over the counter type drugs because they often start with a chemist/pharma company name e.g. things like "Boots vapour chest rub". But in this case it works for all the ones in these code sets.